### PR TITLE
Domain validation

### PR DESCRIPTION
**Pull request by Dermendzhiev**
- add domain validation inside User entity
- handle `DomainException` inside gRPC interceptor
- rename `User` to `UserDataModel`

### CODE REVIEW

**Review by MiroslavStoyanov — DISMISSED** [review]

File: src/microservices/users/src/Api/Interceptors/ServerInterceptor.cs [hunk 1]
```diff
@@ -32,34 +33,44 @@ public ServerInterceptor(ILogger<ServerInterceptor> logger, IMetricsRegistry met
...
+                        {
+                            errorDetail = exception.Message;
+                            statusCode = StatusCode.InvalidArgument;
+                        }
+                        else
+                        {
+                            statusCode = StatusCode.Internal;
+                        }
+
+                        throw new RpcException(new Status(StatusCode.Internal, errorDetail));
```
Comment (Bojo966) [inline src/microservices/users/src/Api/Interceptors/ServerInterceptor.cs]: guessing you want to use the statusCode set earlier

**Review by Bojo966 — DISMISSED** [review]

File: src/microservices/users/src/Api/Interceptors/ServerInterceptor.cs (hunk 1 above)
Comment (Dermendzhiev) [reply src/microservices/users/src/Api/Interceptors/ServerInterceptor.cs]: good point

**Review by MiroslavStoyanov — APPROVED** [review]

**Comment by Bojo966** [comment]
LGTM 🚀

**Comment by Bojo966** [comment]
Don't even test, just merge in prod.

**Comment by Bojo966** [comment]
😮

**Pull request merged**